### DISCUSSION TITLE
Add option to prevent WorkBench data entry for read-only fields

### DISCUSSION THREAD

**Discussion by lexiclevenger** [specify/specify7#5953] [Feature Requests]
**Is your feature request related to a problem? Please describe.**

> Currently, the only option to set a field to read-only is via hard-coding it in the form xml via readonly="true". This does not prevent adding data to this field via a Workbench load.
> It would be good to have an additional field flag in the schema to indicate that a field should be read-only in the form as well as in a Workbench load (validation step?). However, it should not block any back-end data entry via sql scripts and/or API calls.
> 
> Use cases:
> 
> a)        Across our instances we have a number of ‘legacy’ fields where data was added at point of final data migration. We do not want users to add data to any new records in these fields.
> 
> b)       We have some fields that are populated via a server-side sql trigger/API, so we don't want users interacting with it. There is an additional GH ticket request to have this ability to format a field using an aggregator without needing to do anything from the back-end ([#3985](https://github.com/specify/specify7/issues/3985)).

**Describe the solution you'd like**
Add a field flag to prevent data entry through both forms and the WorkBench without blocking back-end data entry.

**Requested by:** Corinna P at CSIRO [(Asana)](https://app.asana.com/0/1201972579963460/1207252626713475/f)

**Comment by CorinnaPaeper** [comment]
That looks good. Thanks!

**Comment by CorinnaPaeper** [comment]
Hi @grantfitzsimmons . This describes what we are after, thanks!
